test/e2e/tls_test.go

```diff
@@ -0,0 +1,517 @@
+package e2e
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/imroc/req/v3"
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestTLSEnabledWithSelfSigned(t *testing.T) {
+	// Path to the helm chart we will test
+	helmChartPath, e := filepath.Abs("../../charts")
+	if e != nil {
+		t.Fatalf(e.Error())
+	}
+	username := "admin"
+	password := "admin"
+
+	namespaceName := "marklogic-" + strings.ToLower(random.UniqueId())
```

Comment:
Namespace should be shorten to avoid FQDN 64 character issue.

Reply:
Tested this FQDN locally with the 64-character limit, and this name works fine.